packages/http/src/validator/index.ts

```diff
@@ -75,17 +76,22 @@ const validateInputBody = (
     checkRequiredBodyIsProvided(requestBody, body),
     E.map(b => [...b, caseless(headers || {})] as const),
     E.chain(([requestBody, body, headers]) => {
+      // parse boundary string from content-type in case media type is multipart/form-data
+      const multipart = require('parse-multipart-data');
+      const contentTypeHeader = headers.get('content-type');
+      const multipartBoundary = contentTypeHeader ? multipart.getBoundary(contentTypeHeader) : "";
+      const mediaType = contentTypeHeader ? contentTypeHeader.replace(new RegExp(";\\s*boundary=" + multipartBoundary), "") : contentTypeHeader;
```

Comment:
After end-end and unit testing, the content-type header appeared in both formats: "multipart/form-data;boundary=--...." and "multipart/form-data; boundary=--...." (with the space after the semicolon). Thus, I'm handling possible whitespace between the media type and the boundary string.

Reply:
1. I'm concerned about your use of a regular expression here.  Notice that the boundary may contain any 7-bit ASCII character, which I think includes many characters with special meanings inside a regular expression (e.g., `.`, `*`, `(`).  Consider using a library to parse the Content-Type header in its entirety.  Here's a very simple example using [whatwg-mimetype](https://github.com/jsdom/whatwg-mimetype):

	```TypeScript
	import MIMEType from 'whatwg-mimetype';
	
	const t = new MIMEType('multipart/form-data; boundary=.*.?.+zzz');
	console.log(t?.essence);
	console.log(t.parameters.get('boundary'));
	```

3. It looks like this block of code is a duplicate of what you added in `mocker/index.ts`.  Could it be factored out?